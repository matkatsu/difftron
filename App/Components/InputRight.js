// @flow
import React from 'react';
import AceEditor from 'react-ace';
import type { Right, InputLanguage, InputTheme } from '../Types';

type EditorSettings = {
  language: InputLanguage,
  theme: InputTheme,
};

type Props = {
  onChange: (input: Right) => void,
  value: Right,
  editorSettings: EditorSettings,
};

const InputRight = ({ onChange, value, editorSettings }: Props) => (
  <AceEditor
    mode={editorSettings.language}
    theme={editorSettings.theme}
    onChange={onChange}
    name="LEFT_INPUT"
    width="100%"
    height="100%"
    value={value}
    showPrintMargin={false}
    editorProps={{ $blockScrolling: true }}
  />
);

export default InputRight;
