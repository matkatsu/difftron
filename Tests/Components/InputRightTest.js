/* eslint-disable import/first */
import '../Setup';
import test from 'ava';
import React from 'react';
import render from 'react-test-renderer';
import InputRight from 'Components/InputRight';

test('InputRight component', (t) => {
  const props = {
    onChange: () => ({}),
    value: 'right input',
    editorSettings: {
      language: 'javascript',
      theme: 'github',
    },
  };
  const tree = render.create(<InputRight {...props} />).toJSON();
  t.snapshot(tree);
});
