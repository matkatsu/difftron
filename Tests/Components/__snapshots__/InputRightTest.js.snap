exports[`InputRight component 1`] = `"{\"__ava_react_jsx\":{\"type\":\"div\",\"props\":{\"id\":\"RIGHT_INPUT\",\"style\":{\"width\":\"100%\",\"height\":\"100%\"}},\"children\":null}}"`;
